package-recipe.el

```diff
@@ -60,6 +61,16 @@
       (format (oref rcp url-format)
               (oref rcp repo))))
 
+(cl-defmethod package-recipe--upstream-protocol ((rcp package-recipe))
+  "Get the protocol used for fetching from the upstream specified in RCP.
+
+If not explicitly specified, assume a local file path."
+  (let* ((url (package-recipe--upstream-url rcp))
+         (type (url-type (url-generic-parse-url url))))
+    (if type
+        type
+      "file")))
```

Comment:
Just use `(or type "file")`.

Reply:
Actually why use variables at all? 

```elisp
(or (url-type
     (url-generic-parse-url
      (package-recipe--upstream-url rcp)))
    "file")
```

Reply:
Oops, you're (obviously) right, though given the switch, it's a moot point now!